Datahub.Core/Services/MSGraphService.cs

```diff
@@ -38,10 +38,18 @@ public class MSGraphService : IMSGraphService
         public async Task<GraphUser> GetUserAsync(string userId, CancellationToken tkn)
         {
             PrepareAuthenticatedClient();
-            var user = await graphServiceClient.Users.Request()
-                .Filter($"id eq '{userId}'")
-                .GetAsync(tkn);
-            return user == null ? null : GraphUser.Create(user[0]);
+            try
+            {
+                var user = await graphServiceClient.Users.Request()
+                    .Filter($"id eq '{userId}'")
+                    .GetAsync(tkn);
+                return user == null ? null : GraphUser.Create(user[0]);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Error fetching user: {userId}: {ex.Message}");
```

Comment:
The exception is being treated here, are you sure the app should continue to function if there is no user?